File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/myshop/given/GivenStage.java
package com.mycompany.myshop.testkit.dsl.port.myshop.given;

import com.mycompany.myshop.testkit.dsl.port.myshop.given.steps.GivenClock;
import com.mycompany.myshop.testkit.dsl.port.myshop.given.steps.GivenCoupon;
import com.mycompany.myshop.testkit.dsl.port.myshop.given.steps.GivenCountry;
import com.mycompany.myshop.testkit.dsl.port.myshop.given.steps.GivenOrder;
import com.mycompany.myshop.testkit.dsl.port.myshop.given.steps.GivenProduct;
import com.mycompany.myshop.testkit.dsl.port.myshop.given.steps.GivenPromotion;
import com.mycompany.myshop.testkit.dsl.port.myshop.then.ThenStage;
import com.mycompany.myshop.testkit.dsl.port.myshop.when.WhenStage;

public interface GivenStage {
    GivenClock clock();

    GivenProduct product();

    GivenPromotion promotion();

    GivenOrder order();

    GivenCountry country();

    GivenCoupon coupon();

    WhenStage when();

    ThenStage then();
}

File: system-test/typescript/src/testkit/dsl/port/myShop/given/given-stage.ts
import type { GivenClock } from './steps/given-clock.js';
import type { GivenProduct } from './steps/given-product.js';
import type { GivenPromotion } from './steps/given-promotion.js';
import type { GivenCoupon } from './steps/given-coupon.js';
import type { GivenCountry } from './steps/given-country.js';
import type { GivenOrder } from './steps/given-order.js';
import type { WhenStage } from '../when/when-stage.js';
import type { ThenStage } from '../then/then-stage.js';

export interface GivenStage {
  clock(): GivenClock;
  product(): GivenProduct;
  promotion(): GivenPromotion;
  coupon(): GivenCoupon;
  country(): GivenCountry;
  order(): GivenOrder;
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}
